packages/medusa/src/strategies/batch-jobs/product/export.ts

```diff
@@ -152,39 +163,20 @@ export default class ProductExportStrategy extends AbstractBatchJobStrategy<
             } as FindProductConfig)
         }
 
-        // Retrieve the highest count of each object to build the dynamic columns later
-        for (const product of products) {
-          const optionsCount = product?.options?.length ?? 0
-          dynamicOptionColumnCount = Math.max(
-            dynamicOptionColumnCount,
-            optionsCount
-          )
-
-          const imageCount = product?.images?.length ?? 0
-          dynamicImageColumnCount = Math.max(
-            dynamicImageColumnCount,
-            imageCount
-          )
-
-          for (const variant of product?.variants ?? []) {
-            if (variant.prices?.length) {
-              variant.prices.forEach((price) => {
-                pricesData.add(
-                  JSON.stringify({
-                    currency_code: price.currency_code,
-                    region: price.region
-                      ? {
-                          currency_code: price.region.currency_code,
-                          name: price.region.name,
-                          id: price.region.id,
-                        }
-                      : null,
-                  })
-                )
-              })
-            }
-          }
-        }
+        const shapeData = this.getProductRelationsDynamicColumnsShape(products)
+        dynamicImageColumnCount = Math.max(
+          shapeData.imageColumnCount,
+          dynamicImageColumnCount
+        )
+        dynamicOptionColumnCount = Math.max(
+          shapeData.optionColumnCount,
+          dynamicOptionColumnCount
+        )
+        dynamicSalesChannelsColumnCount = Math.max(
+          shapeData.salesChannelsColumnCount,
+          dynamicSalesChannelsColumnCount
+        )
```

Comment:
nit(non-blocking): it was hard for me to conclude that `getProductRelationsDynamicColumnsShape` returns 0 for `salesChannelsColumnCount` and as such no sales channel columns would show up when the feature flag is not applied. 

Maybe just add a quick comment here that this is the case?

Reply:
Yeah definitely, this is the case for all dynamically generated column i can do a global comment for them